File: java/1 - tutorial Java/2-variaveis/variaveis.java
// Variaveis em JAVA
/*
String- armazena texto, como "Olá". Os valores da string estão entre aspas duplas
int- armazena inteiros (números inteiros), sem decimais, como 123 ou -123
float- armazena números de ponto flutuante, com decimais, como 19,99 ou -19,99
char- armazena caracteres únicos, como 'a' ou 'B'. Os valores de caracteres estão entre aspas simples
boolean- armazena valores com dois estados: verdadeiro ou falso
*/

 public class variaveis {
    public static void main(String[] args) {
        //variavel do tipo string
        String nome = "Arolnode";
        System.out.println(nome);

        //variavel do tipo number int
        int idade = 45;
        System.out.println(idade);

        //variavel do tipo number double e float
        double money = 2.5;
        Float dinheiro = 2.5f; //é necessario colocar f no final do numero quando é do tipo float
        
        System.out.println(dinheiro+money);

        //variavel do tipo char
        char sala = 'B';
        System.out.println(sala);

        //variavel do tipo boolean
        boolean myBool = true; // é possivel escrever o tipo true ou false
        System.out.println(myBool);

        //Constantes (Final):
        //Você pode criar constantes em Java usando a palavra-chave final. 
        //O valor de uma constante não pode ser alterado uma vez que tenha sido 
        //atribuído.
        final double PI = 3.14159;
    }
}

File: switf/1-fundamentos/variaveis.swift
/*
1. Variáveis e Constantes:
Variáveis são espaços na memória que armazenam valores mutáveis. 
Em Swift, você pode criar uma variável usando var.
*/
var idade = 40
idade = 27;

//Constantes são espaços na memória que armazenam valores imutáveis. 
//Em Swift, você pode criar uma constante usando let.
let nome = "Arolnode"
// nome = "Node" // Isso resultará em um erro, pois não alterar constantes

//Dica: Use constantes sempre que o valor não precisar ser alterado. 
//Isso torna seu código mais seguro e claro.

//Quando se trata de strings, você pode combinar texto usando operador +:
let primeiroNome = "João"
let segundoNome = "Silva"
let nomeCompleto = primeiroNome + " " + segundoNome

//***********************************************************************//
//TIPOS DE DADOS
//Swift é uma linguagem com tipagem estática, o que significa que você precisa 
//declarar o tipo de dado que uma variável ou constante irá armazenar.

//Inteiros: Armazenam números inteiros, positivos e negativos.
var idade: int = 27

//Flutuantes: Armazenam números decimais.
var dinheiro: Double = 1.99

//Booleanos: Armazenam verdadeiro ou falso.
var casado: Bool = true

//Strings: Armazenam texto.
var mundo: String = "Hello, World!!!"

//***********************************************************************//
//Inferência de Tipo:
// Swift possui inferência de tipo, o que significa que em alguns casos o tipo 
//de dado pode ser inferido pelo valor atribuído:

var contador = 0 // Swift infere que é um Int
let nome2 = "Maria" // Swift infere que é uma String

//***********************************************************************//
//Interpolação de Strings:
let nota = 10
let materia = "Fisica"
let mensagem = "Eu tirei \(nota) em \(materia)"

//Isso ajuda a criar strings mais dinâmicas, inserindo valores de variáveis 
//diretamente dentro delas.







